web/src/main/java/org/springframework/security/web/util/matcher/RequestMatchers.java

```diff
@@ -0,0 +1,56 @@
+package org.springframework.security.web.util.matcher;
+
+/**
+ * A factory class to create {@link RequestMatcher} instances.
+ *
+ * @author Christian Schuster
+ * @since 6.1
+ */
+public final class RequestMatchers {
+
+	/**
+	 * Creates a {@link RequestMatcher} that matches if at least one of the
+	 * given {@link RequestMatcher}s matches, if <code>matchers</code>
+	 * are empty then the returned matcher never matches.
+	 * @param matchers the {@link RequestMatcher}s to use
+	 * @return the any-of composed {@link RequestMatcher}
+	 */
+	public static RequestMatcher anyOf(RequestMatcher... matchers) {
+		return request -> {
+			for (RequestMatcher matcher : matchers) {
+				if (matcher.matches(request)) {
+					return true;
+				}
+			}
+			return false;
+		};
+	}
+
+	/**
+	 * Creates a {@link RequestMatcher} that matches if all the
+	 * given {@link RequestMatcher}s match, if <code>matchers</code>
+	 * are empty then the returned matcher always matches.
+	 * @param matchers the {@link RequestMatcher}s to use
+	 * @return the all-of composed {@link RequestMatcher}
+	 */
+	public static RequestMatcher allOf(RequestMatcher... matchers) {
+		return request -> {
+			for (RequestMatcher matcher : matchers) {
+				if (!matcher.matches(request)) {
+					return false;
+				}
+			}
+			return true;
+		};
+	}
+
+	/**
+	 * Creates a {@link RequestMatcher} that matches if the
+	 * given {@link RequestMatcher} does not match.
+	 * @param matcher the {@link RequestMatcher} to use
+	 * @return the inverted {@link RequestMatcher}
+	 */
+	public static RequestMatcher not(RequestMatcher matcher) {
+		return request -> !matcher.matches(request);
+	}
+}
```

Comment:
The project code formatter is going to require a private constructor and possibly a couple of other things. WIll you please run `./gradlew format check` to get the complete list?